From 435afd182e8f5997033da1d69e56094ecb112cad Mon Sep 17 00:00:00 2001
From: Chukun Pan <amadeus@jmu.edu.cn>
Date: Thu, 10 Jun 2025 23:10:16 +0800
Subject: [PATCH] mtd: spi-nand: macronix: disable continuous read for
 MX35LFxGE4AD

In on-die ECC mode, enabling continuous read will cause ubi_io_read error.
[    7.852000] ubi0 warning: ubi_io_read: error -5 while reading ...

So disable it first. Tested on GL.iNet GL-MT3000 with MX35LF2GE4AD flash.

Fixes: 11813857864f ("mtd: spi-nand: macronix: Continuous read support")
Signed-off-by: Chukun Pan <amadeus@jmu.edu.cn>
---
 drivers/mtd/nand/spi/macronix.c | 2 --
 1 file changed, 2 deletions(-)

--- a/drivers/mtd/nand/spi/macronix.c
+++ b/drivers/mtd/nand/spi/macronix.c
@@ -187,7 +187,6 @@ static const struct spinand_info macroni
 		     SPINAND_HAS_QE_BIT,
 		     SPINAND_ECCINFO(&mx35lfxge4ab_ooblayout,
 				     macronix_ecc_get_status),
-		     SPINAND_CONT_READ(macronix_set_cont_read),
 		     SPINAND_READ_RETRY(MACRONIX_NUM_READ_RETRY_MODES,
 					macronix_set_read_retry)),
 	SPINAND_INFO("MX35LF4GE4AD",
@@ -200,7 +199,6 @@ static const struct spinand_info macroni
 		     SPINAND_HAS_QE_BIT,
 		     SPINAND_ECCINFO(&mx35lfxge4ab_ooblayout,
 				     macronix_ecc_get_status),
-		     SPINAND_CONT_READ(macronix_set_cont_read),
 		     SPINAND_READ_RETRY(MACRONIX_NUM_READ_RETRY_MODES,
 					macronix_set_read_retry)),
 	SPINAND_INFO("MX35LF1G24AD",
